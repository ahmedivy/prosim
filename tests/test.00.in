1 5 1
Proc1 6 1 1
DOOP 4
BLOCK 3
DOOP 4
BLOCK 2
DOOP 3
HALT
