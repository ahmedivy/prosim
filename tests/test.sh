#!/bin/sh
export LC_ALL=C

echo ======================================================
echo ====================== TEST $1 =======================
echo ======================================================
if timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.raw; then 
  cat tests/test.$1.raw | sort > tests/test.$1.out
  if diff -b tests/test.$1.out tests/test.$1.expected > /dev/null; then
    if grep "IS_CONCURRENT" tests/test.$1.cfg > /dev/null; then
      for x in {0..100}; do
        if diff tests/test.$1.raw tests/test.$1.out > /dev/null; then
          if [ $x -eq 100 ]; then
            echo FAILED: Output is correct, but no concurrency is apparent
            echo Threads appear to be executed in sequential order
            exit 1
          else 
            echo RETRYING: Output is correct, but no concurrency is apparent
            timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.raw
          fi
        else 
          break
        fi
      done
      #if diff tests/test.$1.raw tests/test.$1.out > /dev/null; then
      #fi
    fi
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ______________Your_output______________ ____________Expected_Output_____________
   #echo +++++++++++++++++++++++++++++++++++++++ ++++++++++++++++++++++++++++++++++++++++
    diff -b -y -W 80 tests/test.$1.out tests/test.$1.expected
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
