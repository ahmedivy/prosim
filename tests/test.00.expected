[01] 00000: process 1 new
[01] 00000: process 1 ready
[01] 00000: process 1 running
[01] 00004: process 1 blocked
[01] 00007: process 1 ready
[01] 00007: process 1 running
[01] 00011: process 1 blocked
[01] 00013: process 1 ready
[01] 00013: process 1 running
[01] 00016: process 1 finished
| 00016 | Proc 01.01 | Run 11, Block 5, Wait 0
